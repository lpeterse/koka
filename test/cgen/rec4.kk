fun foo(xs,n) {
    fun bar(xs : list<int>) {
       match(xs) {
         Cons(_x,xx) -> bar(xx);
         Nil -> n
       }
    };
    bar(xs)
}


fun main() { print(foo([],1)) }
