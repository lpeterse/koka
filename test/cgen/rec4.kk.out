test/cgen/rec4.kk(2,13): warning: xs shadows an earlier local definition or parameter